File: Data Structure/try/build_array.js
class BuildArray{
    constructor(){
        this.length = 0
        this.data = {}
    }

    get(index){
        return this.data[index]
    }

    push(value){
        this.data[this.length] = value
        this.length++
        return this.length
    }

    pop(){
        const lastItem = this.data[this.length-1]
        delete(this.data[this.length-1])
        this.length--
        return lastItem
    }

    // add(value, index){
    //     for(let i = index; i < this.length; i++){
    //         this.data[i+1] = this.data[i]
    //     }
    //     this.data[index] = value
        
    //     this.length++
    //     return this.length
    // }

    delete(index){
        const currentitem = this.data[index]
        this.shiftIndex(index)
        return currentitem
    }

    shiftIndex(index){
        for(let i = index; i < this.length-1; i++){
            this.data[i] = this.data[i+1]
        }
        delete(this.data[this.length-1])
        this.length--
    }


}


const newArray = new BuildArray()
newArray.push(10)
newArray.push(5)
newArray.push(15)
newArray.pop()
newArray.push(25)
newArray.push(35)
newArray.push(45)
newArray.push(55)
newArray.delete(3)
// newArray.add(70, 2)
console.log(newArray)

File: Data Structure/python/build_array.py
class MyArray:
    def __init__(self):
        self.length = 0
        self.data = {}

    def get(self, index):
        return self.data[index]

    def push(self, item):
        self.data[self.length] = item
        self.length += 1
        return self.length

    def pop(self):
        lastItem = self.data[self.length-1]
        del self.data[self.length-1]
        self.length -= 1
        return lastItem

    def delete(self, index):
        item = self.data[index]
        self.shiftItem(index)
        return item

    def shiftItem(self, index):
        for i in range(index,self.length-1):
            self.data[i] = self.data[i+1]
        del self.data[self.length-1]
        self.length -= 1



newArray = MyArray()
newArray.push("Hi")
newArray.push("I am")
newArray.push("Sanusi")
newArray.push("hamed")
newArray.push("tijani")
newArray.push("firdaus")
newArray.push("subomi")
newArray.push("segun")
# newArray.pop()
newArray.delete(6)
print(newArray.data)